scripts/mkdoc.sh

```diff
@@ -54,13 +54,52 @@ if [ ! -d ".venv" ]; then
   echo "Creating virtualenv..."
   ${PYTHON:-python3} -m venv .venv
 
-  # Install sphinx, matplotlib, pandas, scipy, wheel and pydoctor into the venv.
+  # Install sphinx, matplotlib, pandas, scipy, wheel, pydoctor, nbsphinx, pandoc, sphinx-design, ipython and ipywidgets into the venv.
   # doc2dash is optional; it will be installed when -d is given
-  .venv/bin/pip install -q -U pip wheel sphinx==7.4.7 matplotlib pandas scipy pydoctor sphinx-rtd-theme
+  .venv/bin/pip install -q -U pip wheel sphinx==7.4.7 matplotlib pandas scipy pydoctor sphinx-rtd-theme nbsphinx pandoc sphinx-design ipython ipywidgets
 else
   # Upgrade pip in the virtualenv
   echo "Upgrading pip in virtualenv..."
   .venv/bin/pip install -q -U pip wheel
+
+  # Check if nbsphinx is installed and install it if not
+  echo "Checking for nbsphinx, pandoc, sphinx-design and ipython..."
+  if ! .venv/bin/pip show nbsphinx > /dev/null 2>&1; then
```

Comment:
This can be simplified with a `for` loop:

```bash
DEPS="nbsphinx pandoc sphinx-design ipython ipywidgets"
for PKG in $DEPS; do
    if ! .venv/bin/pip show ${PKG} > /dev/null 2>&1; then
        echo "$PKG not found. Installing..."
        .venv/bin/pip install -q $PKG
    else
        echo "$PKG is already installed."
    fi
done
```

But maybe it's easier to just let `pip` do the heavy lfiting:

```
pip install -q -U nbsphinx pandoc sphinx-design ipython ipywidgets
```

Don't spend too much time on this; at some point I will rewrite this whole script using `uv` and a proper virtualenv management system anyway.